### PR TITLE
[fix] Improved test data matches metadata and some housekeeping

### PR DESCRIPTION
**Pull request by jacek-oet**
- Moved the assertion to check that the set of benchmark names in the metadata matches those in the data from home.py to app.py.
- Added a load\_metadata function for reusability and better code organization.
- Updated the Compare page to handle cases where the number of solvers is 1 or 0.

### CODE REVIEW

File: website/app.py [hunk 1]
```diff
@@ -41,5 +47,26 @@
...
+metadata_df = pd.DataFrame(metadata).T.reset_index()
+metadata_df.rename(columns={"index": "Benchmark Name"}, inplace=True)
+# Load the data from the CSV file
+data_url = Path(__file__).parent.parent / "results/benchmark_results.csv"
+raw_df = pd.read_csv(data_url)
+df = raw_df
+
+
+# Assert that the set of benchmark names in the metadata matches those in the data
+csv_benchmarks = set(raw_df["Benchmark"].unique())
```
Comment (siddharth-krishna) [inline website/app.py:62]:
```suggestion
data_df = pd.read_csv(data_url)


# Assert that the set of benchmark names in the metadata matches those in the data
csv_benchmarks = set(data_df["Benchmark"].unique())
```

File: website/compare.py [hunk 2]
```diff
@@ -41,23 +34,36 @@ def load_metadata(file_path):
...
+# Set default solvers based on available solver list length
+if len(solver_list) > 1:
+    default_solver1 = solver_list[0]
+    default_solver2 = solver_list[1]
+elif len(solver_list) == 1:
+    default_solver1 = solver_list[0]
+    default_solver2 = None
+else:
+    st.write("No solvers available for comparison.")
+    st.stop()
```
Comment (siddharth-krishna) [inline website/compare.py:48]:
```suggestion
else:
    st.write("Fewer than 2 solvers available for comparison. Please add more solvers to compare.")
    st.stop()
```
We can also simplify the code below to assume that there are at least 2 solvers.

**Review by siddharth-krishna — APPROVED** [review]
Looks good, thanks! I left some suggestions to simplify the code.

Can we also quickly add something like `streamlit run` to the CI? Just to check that the website loads without errors. Streamlit must have some method to test this without running with a timeout or something. No need to spend too much time on it if there isn't something out-of-the-box. Thanks

**Comment by jacek-oet** [comment]
@siddharth-krishna 
> Can we also quickly add something like streamlit run to the CI? 

Let me research and implement it in new PR.

File: website/compare.py [hunk 3]
```diff
@@ -41,23 +34,35 @@ def load_metadata(file_path):
...
+# Dropdown to select Solver 2 only if there’s more than one solver
+if default_solver2:
+    solver2 = st.selectbox(
+        "Select Solver 2",
+        solver_list,
+        index=solver_list.tolist().index(default_solver2),
+    )
+else:
+    st.write("Only one solver available. Please add more solvers to compare.")
+    st.stop()
```
Comment (siddharth-krishna) [inline website/compare.py:65]: This check and the else block are unnecessary since we check that there are at least 2 solvers above.

**Review by siddharth-krishna — APPROVED** [review]

**Pull request merged**